### PR TITLE
Add Memory Game task

### PR DESCRIPTION
**Pull request by AriadnaKravchuk**
### Memory Game

[Demo](https://ariadnakravchuk.github.io/Memory-Game) |
[Code base](https://github.com/AriadnaKravchuk/Memory-Game)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

File: submissions/AriadnaKravchuk/memory-game/script.js [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+let theme = "",
+    matches = 0,
+    moves = 0;
+let isEndGame = false,
+    secondsCounter = 0,
+    minutesCounter = 0,
+    minutes = "00",
+    seconds = "00";
+let firstCard = [],
+    secondCard = [],
```
Comment (Roophee) [inline submissions/AriadnaKravchuk/memory-game/script.js:26]: Why do you use `array` as default or empty value for this variables? You don't use array in dependence of variables logic.

File: submissions/AriadnaKravchuk/memory-game/script.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        secondCard.removeEventListener("click", flipCard);
+
+        if (matches === CARD_IMG.QUANTITY) {
+            isEndGame = true;
+            scorePopup.innerHTML = `Time: ${minutes}.${seconds} Moves: ${moves}`;
+            setTimeout(() => {
+                popup.classList.remove("hidden");
+            }, TIMEOUT.SHOW_POPUP);
+        }
+        firstCard = [], secondCard = [];
```
Comment (Roophee) [inline submissions/AriadnaKravchuk/memory-game/script.js:157]: Same.  Why are you using `array`? It is truly value and you don't use array or its methods. Maybe `null` or `undefined` will be better choose.

File: submissions/AriadnaKravchuk/memory-game/style.css [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+    --game-page-font-family: 'Gotu', sans-serif;
+    --game-page-board-background: beige;
+    --card-front-background: #bba6c1;
+    --card-back-background: #dbc5db;
+    --status-board-background: #bba6c1;
+    --card-front-border:#c7bac7;
+    --card-back-border: #d1c1d1;
+    --card-front-color: white;
+    --popup-button-color-hover: #a494a8;
+}
```
Comment (Roophee) [inline submissions/AriadnaKravchuk/memory-game/style.css:380]: Please avoid using tag selectors, it is better to use class or id  selectors. This is reason why  >>> [article](https://frontstuff.io/you-need-to-stop-targeting-tags-in-css).
Is `body.sailor-moon` redundant? Can we leave only `.sailor-moon` selector?

File: submissions/AriadnaKravchuk/memory-game/style.css [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+        --card-front-font-size: 3rem;
+        --popup-width: 450px;
+        --popup-title-font-size: 2.6rem;
+        --popup-text-font-size: 1.4rem;
+        --popup-button-padding: 0.938rem;
+        --popup-button-font-size: 1rem;
+    }
+
+    :root body.gothic {
+        --status-board-margin: -1.6rem auto 0;
```
Comment (Roophee) [inline submissions/AriadnaKravchuk/memory-game/style.css:446]: Tag selector issue.

File: submissions/AriadnaKravchuk/memory-game/style.css [hunk 5]
```diff
@@ -0,0 +1,448 @@
...
+    transform: rotateY(180deg); 
+}
+
+.hidden {
+    display: none;
+}
+
+.start-page {
+    background-color:beige;
+}
```
Comment (Roophee) [inline submissions/AriadnaKravchuk/memory-game/style.css:128]: Please, avoid text values of colors, use HEX or RGB style.

File: submissions/AriadnaKravchuk/memory-game/style.css [hunk 6]
```diff
@@ -0,0 +1,448 @@
...
+
+.start-page__inner,
+.popup {
+    font-family: 'Capriola', sans-serif;   
+}
+
+.details__summary,
+.details__content {
+    border: 0.063rem solid silver;
+    background: white;
```
Comment (Roophee) [inline submissions/AriadnaKravchuk/memory-game/style.css:115]: Color naming issue.

**Review by Roophee — COMMENTED** [review]
Hey!!!
You have made great job!!! Pretty design, complex logic - looks like a "professional" good.
Let's make you code better. I have left some comments for you. Please, fix them.

**Comment by AriadnaKravchuk** [comment]
Thank you for the review!
I made changes, now all content is generated in js.
Cards are compared using the date attribute, reordering occurs without order using sorting.

**Comment by Roophee** [comment]
@AriadnaKravchuk  great job!!! You code got more reusable functions and became DRY. It is wonderfull. So you got my approve and your PR is completed!!!
Congrats!!!!

**Pull request merged**